pkg/datapath/linux/config/config.go

```diff
@@ -839,6 +840,17 @@ func getEphemeralPortRangeMin(sysctl sysctl.Sysctl) (int, error) {
 	return ephemeralPortMin, nil
 }
 
+// ignoringEINTR makes a function call and repeats it if it returns an
+// EINTR error.
+func ignoringEINTR(fn func() error) error {
+	for {
```

Comment:
Is there any risk in retrying indefinitely? I think we also have the `resiliency` pkg that already offers some sort of retry functions, but not sure if these APIs ever gained any traction in the codebase...
Also pinging @ti-mo, IIRC he was involved in some offline discussions around improving the handling of EINTR, especially in the context of netlink. Does this go in the right direction or do you have something else in mind?

Reply:
IFAIK it is customary to retry indefinitely when getting EINTR, see e.g.: https://github.com/golang/go/blob/master/src/os/file_posix.go#L242-L256

Reply:
> Also pinging @ti-mo, IIRC he was involved in some offline discussions around improving the handling of EINTR, especially in the context of netlink. Does this go in the right direction or do you have something else in mind?

Any netlink library call can return EINTR, this needs to be fixed in the lib itself.

Reply:
@ti-mo wrote:

> Any netlink library call can return EINTR, this needs to be fixed in the lib itself.

Looks like this is not _actually_ a `EINTR` from a system call, but is mapped from `NLM_F_DUMP_INTR`, set by the kernel when the resource being dumped changes during the dump. This mapping is a recent change in the netlink library, there seems to be an upstream fix for this: https://github.com/vishvananda/netlink/pull/1018. Using this involves retrying the the dump in this specific error condition, IMO.

We can address this error condition by the change in this PR, or by looking for the new (in [#1018](https://github.com/vishvananda/netlink/pull/1018)) error `netlink.ErrDumpInterrupted` instead of `EINTR` after we have updated the lib.

Reply:
Updated the title and description in light of the above information.